### PR TITLE
OAuth via routes are now available

### PR DESCRIPTION
**Pull request by jlaswell**
Most of the logic is crammed into the auth/handlers.go file, so that will need to be cleaned up prior to merge. Handlers were put into the auth package to keep that logic separated at the moment.

User model definite adjustment based on Github user definition. Github does not have a concept of first name or last name, so this has been changed to just name.

auth/client.go is poorly named and should probably house most of the logic in handlers.

### CODE REVIEW

File: auth/handlers.go [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+func createNewGithubUser(githubUser *github.User) {
+	user := User{
+		Username: *githubUser.Login,
+		Name:     *githubUser.Name,
+		Email:    *githubUser.Email,
+	}
+	db, e := database.Connect()
+	defer db.Close()
+	fmt.Printf("%v\n%v\n", db, e)
```
Comment (jlaswell) [inline auth/handlers.go]: I'll need to clean that up.

File: auth/handlers.go [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+	oauthClient := oauthConf.Client(oauth2.NoContext, token)
+	client := github.NewClient(oauthClient)
+	user, _, err := client.Users.Get("")
+	if err != nil {
+		fmt.Printf("client.Users.Get() faled with '%s'\n", err)
+		http.Redirect(w, r, "/", http.StatusTemporaryRedirect)
+		return
+	}
+	// fmt.Printf("Logged in as GitHub user: %s\n", *user.Login)
+	// fmt.Printf("access token set to '%s'\n", token.AccessToken)
```
Comment (jlaswell) [inline auth/handlers.go]: I'll need to clean that up.

**Comment by kyleferguson** [comment]
@jlaswell @larkmullins what do you think about moving all "handlers" to `app/handlers/api` and `app/handlers/web` within more specific files names such as `auth.go`. The handlers can be the same concept as a controller in MVC, in that they're an entry point and by rule of thumb shouldn't contain much logic.

**Comment by jlaswell** [comment]
I like that a lot. It didn't feel right to have them in `auth` by themselves. I'll make that change as part of pulling the auth handler logic into more manageable parts.

**Review by larkmullins — APPROVED** [review]

**Pull request merged**